_minutes/2023-12-07-wecg.md

```diff
@@ -0,0 +1,183 @@
+ * [timothy] I think once a developer is using the theme icons keys, I don't think fallback should happen piecemeal to default icons. That keeps implementation cleaner & self contained. If a dev leaves out light or dark, it will be more obvious for them to fix. If theme icons are missing entirely, fallback to non-theme icons should happen.
+ * [oliver] This isn't theme icons. That's another thing currently supported by Firefox. “dark” and “light” under icons is what we're currently discussing.
+ * [oliver] Example:
+   ```
+     "icons": {
+       "dark": {
+         "128": "icons/128_dark.png"
+       },
+       "light": {
+         "128": "icons/128_light.png"
+       },
+       "128": "icons/128.png"
+     },
+   ```
+ * [carlos] I recently tested icons.dark and icons.light and this prevents an extension from loading in Chrome currently. This issue has been discussed in the group before and we were more leaning towards implementing theme_icons as implemented by Firefox right now. This would allow an implementation that is backwards compatible. Maybe in MV4 we could go with a more streamlined version. If we add something like dark/light, it would be good to support something like SVG from the start.
+ * [timothy] I'd prefer to do SVG support from the start. Backwards compatibility concerns are definitely there when we're reusing the same key. Safari will ignore it, but a JS exception will be thrown if you try to call action.setIcon() in Safari with a non-numeric key. I expect Chrome will be more strict than most with this kind of thing.
+ * [oliver] For clarity, are you supportive of theme_icons or light/dark in
+ * [timothy] The issue is unclear. Preference is to use a new key.
+ * [oliver] Firefox reps, I believe you were planning to remove theme_icons, correct?
+ * [rob] The issue with theme_icons in Firefox is that the meaning of “dark” and “light” deviate from the current conventions on the web platform. So on Firefox's end we are willing to support icons.
+ * [timothy] One potential issue with “icons” is that numeric keys are expected.
+ * [rob] That's my big concern. Firefox also does not support non-numeric keys either and refuses to load the extension/manifest if non-numeric keys are present in “icons”. “Theme_icons” supports dark/light/size and only warns about unrecognized keys (no hard error).
+ * [timothy] We don't currently support anything other than the auto dark mode thing we currently do
+ * [tomislav] Seems there's consensus around the idea of supporting light and dark icons and mutual concerns around backwards compatibility. Suggest we move bikeshedding on the specifics into async conversation in the issue.
+ * [solomon]
+   ```
+   "icons": {
+     "dark": {
+       "icons/128_dark.png"
+     },
+     "light": {
+       "icons/128_light.png"
+     },
+     "icons/128.png"
```

Comment:
This looks strange to me, as it is invalid JSON. Someone else (not me) put this in the meeting notes. @xeenon Do you recall the context around this? Is there anything to fix up?

Reply:
This was an example pasted in the Zoom chat trying to see if that is what I meant to support SVG.

It likely should be the following for the SVG single path case:

```
"icons": {
     "dark": "icons/128_dark.png",
     "light": "icons/128_light.png",
     "*": "icons/128.png"
}
```

Reply:
That looks more sensible, thanks. I'll fix up the notes.